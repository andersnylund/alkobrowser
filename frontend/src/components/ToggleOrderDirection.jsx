import React from 'react';
import { Icon } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { string, func } from 'prop-types';

import Button from './Button';
import { toggleSortAction } from '../actions/filterActions';

export const ToggleOrderDirection = ({ sort, toggleSort }) => {
  return (
    <Button onClick={() => toggleSort()} data-testid="toggle-sort">
      <Icon
        name={`sort content ${sort === 'ASC' ? 'ascending' : 'descending'}`}
      />
    </Button>
  );
};

ToggleOrderDirection.propTypes = {
  sort: string.isRequired,
  toggleSort: func.isRequired,
};

const mapStateToProps = state => ({
  sort: state.filter.sort,
});

const mapDispatchToProps = {
  toggleSort: toggleSortAction,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ToggleOrderDirection);
