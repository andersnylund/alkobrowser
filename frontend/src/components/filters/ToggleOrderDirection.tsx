import React, { FunctionComponent } from 'react';
import { Icon } from 'semantic-ui-react';
import { connect } from 'react-redux';

import Button from '../common/Button';
import { toggleSortAction } from '../../actions/filterActions';
import { AppState } from '../../store';

interface Props {
  sort: string; // TODO: ASC DESC
  toggleSort: () => void;
}

export const ToggleOrderDirection: FunctionComponent<Props> = ({ sort, toggleSort }) => {
  const name = sort === 'ASC' ? 'sort content ascending' : 'sort content descending';
  return (
    <Button onClick={() => toggleSort()} data-testid="toggle-sort-button">
      <Icon name={name} />
    </Button>
  );
};

const mapStateToProps = (state: AppState) => ({
  sort: state.filter.sort,
});

const mapDispatchToProps = {
  toggleSort: toggleSortAction,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ToggleOrderDirection);
