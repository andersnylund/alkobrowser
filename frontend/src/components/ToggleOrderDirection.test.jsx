import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';

import { ToggleOrderDirection } from './ToggleOrderDirection';

describe('<ToggleOrderDirection />', () => {
  afterEach(cleanup);

  it('should render without an error', () => {
    const mockToggleSort = jest.fn();
    render(<ToggleOrderDirection sort="ASC" toggleSort={mockToggleSort} />);
  });

  it('should be clickable and sort ascending', () => {
    const mockToggleSort = jest.fn();
    const { getByTestId } = render(
      <ToggleOrderDirection sort="ASC" toggleSort={mockToggleSort} />
    );
    const button = getByTestId('toggle-sort-button');
    fireEvent.click(button);
    expect(mockToggleSort).toHaveBeenCalled();
    expect(button).toMatchSnapshot();
  });

  it('should sort descending', async () => {
    const mockToggleSort = jest.fn();
    const { getByTestId } = render(
      <ToggleOrderDirection sort="DESC" toggleSort={mockToggleSort} />
    );
    const button = getByTestId('toggle-sort-button');
    expect(button).toMatchSnapshot();
  });
});
