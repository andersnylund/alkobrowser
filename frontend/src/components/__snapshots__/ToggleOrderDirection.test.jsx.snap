// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleOrderDirection /> should be clickable and sort ascending 1`] = `
<button
  class="sc-bdVaJa cHEScy"
  data-testid="toggle-sort-button"
  style="transform: none;"
>
  <i
    aria-hidden="true"
    class="sort content ascending icon"
  />
</button>
`;

exports[`<ToggleOrderDirection /> should sort descending 1`] = `
<button
  class="sc-bdVaJa cHEScy"
  data-testid="toggle-sort-button"
  style="transform: none;"
>
  <i
    aria-hidden="true"
    class="sort content descending icon"
  />
</button>
`;
